### ISSUE TITLE
False positive within typespecs on NestedFunctionCalls code readability check 

### ISSUE THREAD

**Issue by Eiji7** [rrrene/credo#1170]
### Environment

* Credo version (`mix credo -v`):
```
$ mix credo -v
1.7.10-ref.(HEAD odłączone na origin/master).2e15cdd2
```
* Erlang/Elixir version (`elixir -v`):
```
$ elixir -v
Erlang/OTP 27 [erts-15.2] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit:ns]

Elixir 1.18.0-rc.0 (f00f759) (compiled with Erlang/OTP 27)
```
* Operating system:
> Gentoo Linux using `desktop/plasma` profile for version `23.0` and with kernel version `6.6.41`.

### What were you trying to do?

I have simply tried to use non-literal as `Keyword`'s value. In my case it's user-defined type with some argument.

```elixir
@type type_name :: Keyword.t(Some.remote(some_arg))
```

### Expected outcome

This check should not appear for all [Typespec attributes | Module @ Elixir documentation](https://hexdocs.pm/elixir/Module.html#module-typespec-attributes).

### Actual outcome

```
Checking (…) source files ...

  Code Readability                                                                                                               
┃ 
┃ [R] → Use a pipeline instead of nested function calls.
┃       (…)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.1 seconds (0.04s to load, 0.1s running 102 checks on (…) files)
(…) mods/funs, found 1 code readability issue.

Use `mix credo explain` to explain issues, `mix credo --help` for options.
```

**Comment by rrrene** [comment]
Thx for reporting! :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by Eiji7** [comment]
@rrrene This fixes:
https://gitlab.com/ramurix-software/dev-joy/-/blob/v2.0.0/lib/dev_joy/api/item.ex?ref_type=tags#L105
but **not**:
https://gitlab.com/ramurix-software/dev-joy/-/blob/v2.0.0/lib/dev_joy/generator/part.ex#L49

**Comment by Eiji7** [comment]
Oh, looks like you have added function clause for only `@type` attribute, right? There should be a `when` guard like `when attr_name in ~w[callback macrocallback opaque spec type typep]a`.

**Comment by Eiji7** [comment]
@rrrene Before pushing it I have verified locally that both lines are no longer emit said false positive `credo` message.
